src/components/buttons/_private/ButtonBase/ButtonBase.tsx

```diff
@@ -39,6 +39,8 @@ export interface IButtonCommonProps extends ILocalContainerProps {
 	onClick?: FunctionGeneric;
 	/** The html element tag used for the button. */
 	tag?: string;
+	/** Props specific to the tag prop tag defined and applied to this component. */
+	tagProps?: {[prop: string]: any};
```

Comment:
The `href` below was failing because it's not a valid Button prop:
```<Button tag='a' href='https://yoyo.com' />```
Rather than add various props for any number of common tag types (e.g. <a href />) or allowing any prop to be set and lose typing support for invalid props, `tagProps` has been created to handle props for a specific tag. This is a more type-friendly and maintainable approach than the alternatives.